experimental/Schemes/SpecOpen.jl

```diff
@@ -975,7 +1004,7 @@ function maximal_extension(
 end
 
 ### the restriction of a morphism to open subsets in domain and codomain
-function restriction(
```

Comment:
In line 888 we have
`function restrict(f::SpecOpenMor, U::SpecOpen, V::SpecOpen; check::Bool=true)`
so this seems to overwrite the other one.
As far as I can tell each function works sometimes but not always.
Please debug @HechtiDerLachs 
(I tried but failed in reasonable time)

Reply:
That's weird. By now, they both seem to do the same thing and the code below can go. Probably this had sneaked in unnoticed with a faulty merge? Either way: We should be OK now.

Edit: The debugging has already happened? Or are there still issues?